pkg/synthetictests/static_pod.go

```diff
@@ -88,22 +94,25 @@ func testStaticPodLifecycleFailure(events monitorapi.Intervals, kubeClientConfig
 		}
 	}
 
-	// now check each failure to see if we eventually reached the level.  If we eventually reached the level, then don't flag it
+	// now check each failure to see if we eventually reached the level.  If we eventually reached the level or greater,
+	// then don't flag it
 	for _, staticPodFailure := range staticPodFailures {
 		events, err := kubeClient.EventsV1().Events(staticPodFailure.namespace).List(ctx, metav1.ListOptions{})
 		if err != nil {
 			failures = append(failures, err.Error())
 			continue
 		}
-
 		for _, event := range events.Items {
 			isRevisionUpdate := event.Reason == "NodeCurrentRevisionChanged"
 			isForNode := strings.Contains(event.Note, staticPodFailure.node)
-			reachedRevision := strings.Contains(event.Note, " to "+staticPodFailure.revision+" because static pod is ready")
-			if isRevisionUpdate && isForNode && reachedRevision {
-				// if we reach the level eventually, don't fail the test. We might choose to add an "it's slow" test, but
-				// it hasn't failed.
-				continue
+			matches := regexp.MustCompile("to ([0-9]+) because static pod is ready").FindStringSubmatch(event.Note)
+			if len(matches) == 2 {
+				reachedRevision, _ := strconv.ParseInt(matches[1], 0, 64)
+				if isRevisionUpdate && isForNode && reachedRevision >= staticPodFailure.revision {
+					// if we reach the level eventually, don't fail the test. We might choose to add an "it's slow" test, but
+					// it hasn't failed.
+					continue
+				}
 			}
 
 			failures = append(failures, staticPodFailure.failureMessage)
```

Comment:
oh, here it is.  we are iterating every event.  We're going to have events that are for prior revisions, but if we manage to find one that indicates "we reached this level" then we don't want to add the failure.  The overall for loop needs to have a `foundEventForProperRevision` marker that is check instead of adding failures here.

Reply:
Will fix. I had convinced myself this wasn't happening but you are right.